assets/activation-scripts/activate.d/zdotdir/.zprofile

```diff
@@ -23,11 +23,12 @@ restore_saved_vars() {
     export FLOX_ORIG_ZDOTDIR="$_save_FLOX_ORIG_ZDOTDIR"
     export ZDOTDIR="$_save_ZDOTDIR"
     export _activate_d="$_save_activate_d"
+    export _flox_activate_tracer="$_save_flox_activate_tracer"
+    export _flox_env_helper="$_save_flox_env_helper"
     export FLOX_ZSH_INIT_SCRIPT="$_save_FLOX_ZSH_INIT_SCRIPT"
     export _FLOX_ACTIVATION_STATE_DIR="$_save_FLOX_ACTIVATION_STATE_DIR"
-    export _FLOX_RESTORE_PATH="$_save_FLOX_RESTORE_PATH"
-    export _FLOX_RESTORE_MANPATH="$_save_FLOX_RESTORE_MANPATH"
     export _FLOX_ACTIVATION_PROFILE_ONLY="$_save_FLOX_ACTIVATION_PROFILE_ONLY"
+    source =("$_flox_env_helper" "zsh")
```

Comment:
Looks like we're running _flox_env_helper 10 times for a single activation. That seems like a lot. Can we just do it twice like we are for other shells?

Reply:
The reasoning is to defend against breakage of PATH or MANPATH from any file in ZDOTDIR. If we can refactor this to be just twice and it's passing tests, that may be okay. We need to make sure that before we run anything that depends on PATH or MANPATH ourselves, we do cleanup